kfs-core/src/main/java/edu/arizona/kfs/gl/batch/service/impl/BudgetAdjustmentCashTransferServiceImpl.java

```diff
@@ -0,0 +1,596 @@
+     */
+    @Override
+    public void generateBudgetAdjustmentCashTransferTransactions() {
+    	LOG.debug("generateBudgetAdjustmentCashTransferTransactions() started");
+        
+        Date executionDate = dateTimeService.getCurrentSqlDate();
+        Date runDate = new Date(runDateService.calculateRunDate(executionDate).getTime());
+        int reportBudgetAdjustDocLoaded = 0;
+        int reportOriginEntryGenerated = 0;
+    	   	
+    	 try {
+    		 PrintStream OUTPUT_GLE_FILE_ps = new PrintStream(batchFileDirectoryName + File.separator + GeneralLedgerConstants.BatchFileSystem.CASH_TRANSFER_TRANSACTIONS_OUTPUT_FILE + GeneralLedgerConstants.BatchFileSystem.EXTENSION);
+    		     		              
+             //retrieve unique document numbers list             
+             List<String> docNumberList = retrieveUniqueDocumentNumbersList();            
+             
+             // don't try to create an entries file or a done file if no budget adjustment transactions were found.
+             if (ObjectUtils.isNotNull(docNumberList) && !docNumberList.isEmpty()) {
+            	 reportBudgetAdjustDocLoaded = docNumberList.size();
+            
+            	 //now that we have a list of the document numbers, generate cash transfer transactions for each document number in file
+            	 reportOriginEntryGenerated = generateCashTransferGeneralLedgerEntries(docNumberList, runDate, OUTPUT_GLE_FILE_ps);
+            	 
+            	 // create a done file
+                 String budgetAdjustmentCashTransferDoneFileName = batchFileDirectoryName + File.separator + GeneralLedgerConstants.BatchFileSystem.CASH_TRANSFER_TRANSACTIONS_OUTPUT_FILE + GeneralLedgerConstants.BatchFileSystem.DONE_FILE_EXTENSION;
+                 File doneFile = new File (budgetAdjustmentCashTransferDoneFileName);
+                 if (!doneFile.exists()) {
+                     try {
+                         doneFile.createNewFile();
+                     } catch (IOException e) {
+                         throw new RuntimeException("Error creating cash transfer done file", e);
+                     }
+            	 
+                 }                
+                        
+             }
+           
+             //delete all budget adjustment transactions  
+             budgetAdjustmentTransactionDao.deleteAllBudgetAdjustmentTransactions();
+             OUTPUT_GLE_FILE_ps.close();
+             reportWriterService.writeStatisticLine("GLBA TRANSACTIONS LOADED        (GL_BUDGET_ADJUST_TRN_T) %,9d", reportBudgetAdjustDocLoaded);
+             reportWriterService.writeStatisticLine("CASH TRANSFER TRANSACTIONS GENERATED                     %,9d", reportOriginEntryGenerated);             
+
+         }
+         catch (FileNotFoundException e) {
+             throw new RuntimeException("generateBudgetAdjustmentCashTransferTransactions Stopped: " + e.getMessage(), e);
+         }
```

Comment:
We should also catch IOException possibly generated at the  OUTPUT_GLE_FILE_ps.close(); statement and throw a RuntimeException